File: src/Enterprise/C#/patterns/Observer.cs
using System;
using System.Linq;
namespace Genkidama.PatternExamples;
public static class ObserverExample { public static bool Run(){Func<int,string>[] obs={i=>$"audit:{i}",i=>$"dashboard:{i}"};return string.Join('>',obs.Select(o=>o(42)))=="audit:42>dashboard:42";} }

File: src/Enterprise/VB.NET/patterns/Observer.vb
Imports System
Imports System.Linq
Friend Module ObserverExample
    Friend Function Run() As Boolean
        Dim observers As Func(Of Integer,String)()={Function(i) $"audit:{i}",Function(i) $"dashboard:{i}"}
        Return String.Join(">",observers.Select(Function(o) o(42)))="audit:42>dashboard:42"
    End Function
End Module
